pydoctor/astbuilder.py

```diff
@@ -47,28 +45,142 @@ def node2fullname(expr: Optional[ast.expr], ctx: model.Documentable) -> Optional
     dottedname = node2dottedname(expr)
     if dottedname is None:
         return None
-    base = ctx.expandName(dottedname[0])
-    if base:
-        return '.'.join([base] + dottedname[1:])
+    return ctx.expandName('.'.join(dottedname))
+
+
+def _maybeAttribute(cls: model.Class, name: str) -> bool:
+    """Check whether a name is a potential attribute of the given class.
+    This is used to prevent an assignment that wraps a method from
+    creating an attribute that would overwrite or shadow that method.
+
+    @return: L{True} if the name does not exist or is an existing (possibly
+        inherited) attribute, L{False} otherwise
+    """
+    obj = cls.find(name)
+    return obj is None or isinstance(obj, model.Attribute)
+
+
+def _handleAliasing(
+        ctx: model.CanContainImportsDocumentable,
+        target: str,
+        expr: Optional[ast.expr]
+        ) -> bool:
+    """If the given expression is a name assigned to a target that is not yet
+    in use, create an alias.
+    @return: L{True} iff an alias was created.
+    """
+    if target in ctx.contents:
+        return False
+    full_name = node2fullname(expr, ctx)
+    if full_name is None:
+        return False
+    ctx._localNameToFullName_map[target] = full_name
+    return True
+
+
+def bind_args(call: ast.Call, sig: Signature) -> BoundArguments:
```

Comment:
I agree it would make more sense to put the signature first.

I had to look up what a `None` keyword means, so indeed that needs a comment: `None` means `**kwargs` was used in the call, in which case there is only a dictionary value and no name associated with it.

The helper is there to bridge the interface of `ast.Call` to `inspect.Signature`: although they are similar, their interfaces don't align exactly.